app/src/main/java/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSender.java

```diff
@@ -34,6 +35,16 @@ public class ReportStreamLabOrderSender implements LabOrderSender {
 
     private static final String CLIENT_NAME = "flexion.etor-service-sender";
 
+    private String cachedPrivateKey;
+
+    protected synchronized String getCachedPrivateKey() {
+        return cachedPrivateKey;
+    }
+
+    protected synchronized void setCachedPrivateKey(String cachedPrivateKey) {
```

Comment:
Is the intent that setCachedPrivateKey is only ever called once per execution? If so, instead of synchronizing on get and set, we may need a way to synchronize between and including both calls.

Reply:
@JohnNKing you mean to have a lock for both methods?

Reply:
Potentially -- or a single synchronized method that calls set & get (e.g. retrievePrivateKey()). Let's see if I can map out a scenario, as that may help:

Without anything being synchronized:
1. Thread 1: getCachedPrivateKey() called => null returned
2. Thread 2: getCachedPrivateKey() called => null returned
3. Thread 1: setCachedPrivateKey() called => (method still executing...)
4. Thread 2: setCachedPrivateKey() called => this.cachedPrivateKey set to key value and returns
5. Thread 1: this.cachedPrivateKey set to key value and returns

With get and set methods synced, the above can't happen. However, this can:
1. Thread 1: getCachedPrivateKey() called => null returned
2. Thread 2: getCachedPrivateKey() called => null returned
3. Thread 1: setCachedPrivateKey() called => this.cachedPrivateKey set to key value and returns
4. Thread 2: setCachedPrivateKey() called => this.cachedPrivateKey set to key value and returns

Thread 2 is prevented from calling set or get if Thread 1 is doing the same, but what about when Thread 1 is between those calls? (e.g. what if line 121: key = secrets.getKey(senderPrivateKey) takes a long time to execute?)